src/core/meshcop/dataset_local.cpp

```diff
@@ -195,8 +195,15 @@ otError DatasetLocal::Get(otOperationalDataset &aDataset) const
         case Tlv::kNetworkName:
         {
             const NetworkNameTlv *tlv = static_cast<const NetworkNameTlv *>(cur);
-            memcpy(aDataset.mNetworkName.m8, tlv->GetNetworkName(), tlv->GetLength());
-            aDataset.mNetworkName.m8[tlv->GetLength()] = '\0';
+            uint8_t length = tlv->GetLength();
+
+            if (length > sizeof(aDataset.mNetworkName) - 1)
```

Comment:
Shall we treat this situation as invalid data?

Reply:
This information is being read from flash.  Not obvious to me whether we should try to use what can be read or simply fail.  If the latter, the device will need to be recommissioned.